packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -224,11 +222,11 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
                             this.addedGCOutboundReference(srcHandle, outboundHandle),
                         tree.trees[path]);
                     // This is the case of rehydrating a detached container from snapshot. Now due to delay loading of
-                    // data store, if the data store is loaded after the container is attached, then we missed marking
-                    // the channel as attached. So mark it now. Otherwise add it to local channel context queue, so
-                    // that it can be mark attached later with the data store.
+                    // data store, if the data store is loaded after the container is attached, then we missed making
+                    // the channel visible. So do it now. Otherwise, add it to local channel context queue, so
+                    // that it can be make it visible later with the data store.
                     if (dataStoreContext.attachState !== AttachState.Detached) {
-                        (channelContext as LocalChannelContextBase).markAttached();
+                        (channelContext as LocalChannelContextBase).makeVisible();
```

Comment:
Isn't this breaking compat? context can be from N-2 version.
Also, we should figure out how not to need casting here.

Reply:
No, this is calling down into the channel contexts for DDSs not up to the data store context.